torch/csrc/distributed/c10d/control_collectives/ControlCollectives.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <ATen/core/ivalue.h>
+#include <chrono>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include <c10/macros/Macros.h>
+#include <torch/custom_class.h>
+
+namespace c10d {
+
+using namespace std::chrono_literals;
+
+class TORCH_API ControlCollectives : public torch::CustomClassHolder {
+ public:
+  virtual void barrier(
+      const std::string& key,
```

Comment:
`key` is better, I wonder if it is actually a 'token'?

Reply:
I think they're equivalent -- key maps better to the underlying concept so will just leave it as is